polynote-frontend/polynote/ui.js

```diff
@@ -1349,35 +1373,23 @@ export class MainUI extends EventTarget {
             }
         });
 
-        this.toolbarUI.addEventListener('RunAll', () =>
-            this.runCells(this.currentNotebook.querySelectorAll('.cell-container.code-cell'))
-        );
-
-        this.toolbarUI.addEventListener('RunToCursor', () => {
-            const cells = [...this.currentNotebook.querySelectorAll('.cell-container.code-cell')];
-            const activeCell = Cell.currentFocus.container;
-
-            const activeCellIndex = cells.indexOf(activeCell);
-            if (activeCellIndex < 0) {
-                console.log("Active cell is not part of current notebook?");
-                return;
+        this.toolbarUI.addEventListener('RunAll', (evt) => {
+            if (this.notebookUI) {
+                console.log("forwarding to", this.notebookUI)
+                evt.forward(this.notebookUI);
             }
-
-            this.runCells(cells.slice(0, activeCellIndex + 1));
         });
 
-        this.toolbarUI.addEventListener('RunCurrentCell', () => {
-            const cells = [...this.currentNotebook.querySelectorAll('.cell-container.code-cell')];
-            const activeCell = Cell.currentFocus.container;
-
-            const activeCellIndex = cells.indexOf(activeCell);
-
-            if (activeCellIndex < 0) {
-                console.log("Active cell is not part of current notebook?");
-                return;
+        this.toolbarUI.addEventListener('RunToCursor', (evt) => {
+            if (this.notebookUI) {
+                evt.forward(this.notebookUI);
```

Comment:
Hmm, interesting. I was also thinking about how we could better handle this pattern of thing-needs-events-from-its-sibling/cousin/whatever.

Reply:
This was just a dumb helper which I'm not even sure is necessary - it's basically just `evt.forward(foo)` instead of `foo.dispatch(evt)` with not much else... 

There's probably a better way to do this...

Reply:
Was thinking maybe a child should have a way of saying it wants certain event types from its ancestor? So the event handling logic for, say, `RunToCursor` would be in `NotebookUI` and it wouldn't even be mentioned directly in `MainUI`? And `UIEventTarget` would know based on something that `NotebookUI` defines that it should propagate `RunToCursor` events to `NotebookUI`?

I dunno, maybe it's too magical... but it does kinda bother me how we have to have a bunch of delegation boilerplate in `MainUI` so that its children can get stuff from the toolbar (and similar situations)

Reply:
I guess one solution could be, if we don't care at all about the hierarchy, to have a global event bus that anything could register/send events on. Maybe that would be simpler?